bigquery/google/cloud/bigquery/dataset.py

```diff
@@ -197,353 +254,255 @@ class Dataset(object):
 
-        :rtype: list of :class:`~google.cloud.bigquery.dataset.AccessEntry`
-        :returns: a list of parsed entries.
-        :raises: :class:`ValueError` if a entry in ``access`` has more keys
-                 than ``role`` and one additional key.
-        """
-        result = []
-        for entry in access:
-            entry = entry.copy()
-            role = entry.pop('role', None)
-            entity_type, entity_id = entry.popitem()
-            if len(entry) != 0:
-                raise ValueError('Entry has unexpected keys remaining.', entry)
-            result.append(
-                AccessEntry(role, entity_type, entity_id))
-        return result
-
-    def _set_properties(self, api_response):
-        """Update properties from resource in body of ``api_response``
-
-        :type api_response: dict
-        :param api_response: response returned from an API call.
+        Returns:
+            Dict[str, object]: The dataset represented as an API resource
         """
-        self._properties.clear()
-        cleaned = api_response.copy()
-        access = cleaned.pop('access', ())
-        self.access_entries = self._parse_access_entries(access)
-        if 'creationTime' in cleaned:
-            cleaned['creationTime'] = float(cleaned['creationTime'])
-        if 'lastModifiedTime' in cleaned:
-            cleaned['lastModifiedTime'] = float(cleaned['lastModifiedTime'])
-        if 'defaultTableExpirationMs' in cleaned:
-            cleaned['defaultTableExpirationMs'] = int(
-                cleaned['defaultTableExpirationMs'])
-        if 'labels' not in cleaned:
-            cleaned['labels'] = {}
-        self._properties.update(cleaned)
-
-    def _build_access_resource(self):
-        """Generate a resource fragment for dataset's access entries."""
-        result = []
-        for entry in self.access_entries:
-            info = {entry.entity_type: entry.entity_id}
-            if entry.role is not None:
-                info['role'] = entry.role
-            result.append(info)
-        return result
-
-    def _build_resource(self):
-        """Generate a resource for ``create`` or ``update``."""
-        resource = {
-            'datasetReference': {
-                'projectId': self.project, 'datasetId': self.dataset_id},
-        }
-        if self.default_table_expiration_ms is not None:
-            value = self.default_table_expiration_ms
-            resource['defaultTableExpirationMs'] = value
-
-        if self.description is not None:
-            resource['description'] = self.description
-
-        if self.friendly_name is not None:
-            resource['friendlyName'] = self.friendly_name
+        return copy.deepcopy(self._properties)
 
-        if self.location is not None:
-            resource['location'] = self.location
+    def _build_resource(self, filter_fields):
+        """Generate a resource for ``update``."""
+        partial = {}
+        for f in filter_fields:
```

Comment:
Nit: We don't use 1-letter variable names. How about `field` or `field_name` instead of `f`?